File: snippets/multiple-files-single-thread/write-file/java/basic.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;

// Write to file (creates or overwrites) - Java 11+
Files.writeString(Paths.get("file.txt"), "Hello, World!");

// Append to file
Files.writeString(Paths.get("file.txt"), "\nNew line", StandardOpenOption.APPEND);

// Write multiple lines
List<String> lines = List.of("Line 1", "Line 2", "Line 3");
Files.write(Paths.get("file.txt"), lines);

File: snippets/multiple-files-single-thread/write-file/python/basic.py
# Write to file (creates or overwrites)
with open("file.txt", "w") as file:
    file.write("Hello, World!")

# Append to file
with open("file.txt", "a") as file:
    file.write("\nNew line")

# Write multiple lines
lines = ["Line 1", "Line 2", "Line 3"]
with open("file.txt", "w") as file:
    file.writelines(line + "\n" for line in lines)
